File: flutter Application/App/lib/models/lost_found.dart
class LostFound {
  final String reportId;
  final String type;
  final String imageUrl;
  final String description;
  final String status;

  LostFound({
    required this.reportId,
    required this.type,
    required this.imageUrl,
    required this.description,
    required this.status,
  });

  factory LostFound.fromJson(Map<String, dynamic> json) => LostFound(
    reportId: json["reportId"],
    type: json["type"],
    imageUrl: json["imageUrl"],
    description: json["description"],
    status: json["status"],
  );
}

File: backend/models/LostFound.js
// // models/LostFound.js

const mongoose = require("mongoose");

const lostFoundSchema = new mongoose.Schema(
  {
    type: {
      type: String,
      required: true,
      enum: ["lost", "found"],
    },
    item: {
      type: String,
      required: true,
    },
    description: {
      type: String,
    },
    status: {
      type: String,
      default: "open",
      enum: ["open", "claimed", "resolved"],
    },
    reportedAt: {
      type: Date,
      default: Date.now,
    },
    location: {
      type: String,
      required: true,
    },
    coordinates: {
      lat: {
        type: Number,
        required: true,
      },
      lng: {
        type: Number,
        required: true,
      },
    },
    reporter: {
      type: String,
    },
    imageUrl: {
      type: String,
    },
    claimedBy: {
      type: String,
    },
    claimedAt: {
      type: Date,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("LostFound", lostFoundSchema);
